lib/src/easy_localization_app.dart

```diff
@@ -17,6 +18,9 @@ class EasyLocalization extends StatefulWidget {
   final String path;
   final AssetLoader assetLoader;
   final bool saveLocale;
+  final Widget preloaderWidget;
+  final Color preloaderColor;
+  final VoidCallback onLocaleChange;
```

Comment:
@Overman775 i think should remove this `preloaderWidget` and `preloaderColor` , make it internel 
why needs to `onLocaleChange`

Reply:
preloaderColor need to be left, design may be different, for example a dark theme, this will help smooth the transition and avoid flickering

Reply:
@Overman775 ok about `preloaderColor` wat about `onLocaleChange``

Reply:
You can remove or leave, as you wish, 50/50. You can also add a setter, this will allow you to monitor the locale

Reply:
great work i think i'll update this code

Reply:
@Overman775  savedlocale not working ...

Reply:
@Overman775 something wrong with `SharedPreferences` store `locale.countryCode` alyaws return `US`
output : `ar_DZ`
```
[log] easy localization: Set locale ar_DZ
[log] easy localization: Locale saved ar_DZ
[log] easy localization: initSavedAppLocale ar_US 
```
output : `ru_RU`
```
[log] easy localization: Set locale ru_RU
[log] easy localization: Locale saved ru_RU
[log] easy localization: initSavedAppLocale ru_US 
```

Reply:
Need change the region in the iPhone settings, it is below the language

Reply:
@Overman775  no this on android plz chek #71

Reply:
@Overman775 this issue get when changed init savelocle men new code  
```dart
if (_savedLocale != null && this.saveLocale) {
      log('easy localization: Saved locale loaded ${_savedLocale.toString()}');
      locale = _savedLocale;
    }
```
`_savedLocale` always ret null

Reply:
On real device all fine...
@aissat From what part of the code is the log easy localization: initSavedAppLocale?

Reply:
@aissat develop branch?

Reply:
@Overman775 add 

`log('easy localization: initSavedAppLocale ${_strLocale.toString()}');` in `initSavedAppLocale()`

Reply:
> @aissat develop branch?

yup

Reply:
```
[log] easy localization: Device locale ru_RU
[log] easy localization: initSavedAppLocale ar_DZ
[log] easy localization: Saved locale loaded ar_DZ
[log] easy localization: Set locale ar_DZ
[MyApp# locale] ar_DZ
[MyApp# Intl.defaultLocale] ar_DZ
[log] easy localization: Locale saved ar_DZ
```

Reply:
```
  static Future initSavedAppLocale() async {
    SharedPreferences _preferences = await SharedPreferences.getInstance();
    var _codeLang = _preferences.getString('codeLa');
    var _codeCoun = _preferences.getString('codeCa');

    _savedLocale = _codeLang != null ? Locale(_codeLang, _codeCoun) : null;
    log('easy localization: initSavedAppLocale ${_savedLocale.toString()}');
  }
```

Reply:
@aissat hmm.... try run deleteSaveLocale()... maybe saved broken due to previous bug

Reply:
first run 👍 
```
[log] easy localization: Set locale en_US
[log] easy localization: Device locale en_US
[log] easy localization: initSavedAppLocale ar_US
[MyApp# locale] en_US
[MyApp# Intl.defaultLocale] en_US
[log] easy localization: Locale saved en_US
[LanguageView] ar_DZ
[log] easy localization: Set locale ar_DZ
[MyApp# locale] ar_DZ
[MyApp# Intl.defaultLocale] ar_DZ
[MyApp# locale] ar_DZ
[MyApp# Intl.defaultLocale] ar_DZ
[log] easy localization: Locale saved ar_DZ
```
reload app
```
[log] easy localization: Set locale en_US
[log] easy localization: Device locale en_US
[log] easy localization: initSavedAppLocale ar_US
[MyApp# locale] en_US
[MyApp# Intl.defaultLocale] en_US
[log] easy localization: Locale saved en_US

```

Reply:
@aissat android version?

Reply:
android Q

Reply:
check my `develop` branch

Reply:
@aissat check develop and dev branch, all fine, no bug

Reply:
@aissat await _preferences.clear(); clean all prefences wery bad way, becouse others packets can use shared prefences, need clean only yours vals

Reply:
@aissat Why `[log] easy localization: Set locale en_US` first? You add somthing?

Reply:
> @aissat await _preferences.clear(); clean all prefences wery bad way, becouse others packets can use shared prefences, need clean only yours vals

and sharedPrefences name for locale bad, easy_locale look beter, and others paccket 100% not owerwrite this val